server.js

```diff
@@ -1,29 +1,95 @@
-import express from 'express'
-import bodyParser from 'body-parser'
-import cors from 'cors'
-import mongoose from 'mongoose'
+import express from "express";
+import bodyParser from "body-parser";
+import cors from "cors";
+import mongoose from "mongoose";
 
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughts"
-mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
-mongoose.Promise = Promise
+const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughts";
+mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true });
+mongoose.Promise = Promise;
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
+const Thought = mongoose.model("Thought", {
+  message: {
+    type: String,
+    required: true,
+    minlength: 5,
+    maxlength: 140,
+  },
+  hearts: {
+    type: Number,
+    default: 0,
+    required: true,
+  },
+  createdAt: {
+    type: Date,
+    default: Date.now,
+  },
+});
+
+// Defines the port the app will run on. Defaults to 8080, but can be
 // overridden when starting the server. For example:
 //
 //   PORT=9000 npm start
-const port = process.env.PORT || 8080
-const app = express()
+const port = process.env.PORT || 8080;
+const app = express();
+
+const listEndpoints = require("express-list-endpoints");
 
 // Add middlewares to enable cors and json body parsing
-app.use(cors())
-app.use(bodyParser.json())
+app.use(cors());
+app.use(bodyParser.json());
 
 // Start defining your routes here
-app.get('/', (req, res) => {
-  res.send('Hello world')
-})
+app.get("/", (req, res) => {
+  res.send(listEndpoints(app));
+});
+
+//Get the 20 latest thoughts route, desc = newest first
+app.get("/thoughts", async (req, res) => {
+  const thoughts = await Thought.find()
+    .sort({ createdAt: "desc" })
+    .limit(20)
+    .exec();
+  res.json(thoughts);
+});
+
+// Post/add new thought
+app.post("/thoughts", async (req, res) => {
+  // Retrive the information sent by the client to our API endpoint
+  const { message } = req.body;
+  // Then use our mongoose model to create the database entry
+  const thought = new Thought({ message });
+
+  try {
+    // Success-case, send good status code to the client
+    const savedThought = await thought.save({ message });
+    res.status(200).json(savedThought);
```

Comment:
Since you put the `message` in the `thought` on line 60, you don't need it on row 64 as well:
```
const thought = new Thought({ message });
  try {
    const savedThought = await thought.save();
    res.status(200).json(savedThought);
  }
```